cli/index.js

```diff
@@ -39,13 +43,355 @@ const signUp = (args) => {
+
+    const { userId, password, confirmPassword, username, firstName, 
+    lastName, email, bio, website, location, picture } = args;
+
+    const updateUserData = {
+        userId,
+        password,
+        confirmPassword,
+        username,
+        firstName,
+        lastName,
+        email,
+        bio,
+        website,
+        location,
+        picture,
+        Birthday
+    };
+
+    console.log(chalk.yellow.bold("updating..."));
+
+    axios
+        .put("/user/{username}", updateUserData)
+        .then((res) => {
+            console.log(res.data);
+
+            const FBIdToken = `Bearer ${res.data.token}`;
+            // TODO: create a file in the user's machine to store the auth token
+            console.log(chalk.green.bold(" update successfully ✔️"));
+        })
+        .catch((error) => {
+            console.log(chalk.red.bold(`Could not update this account -> code: ${error.response.statusText}, message: ${JSON.stringify(error.response.data)}`));
+        });
+}
+//get post command definition
+const getpost = (args) => {
+
+    console.log(chalk.yellow.bold("getting post..."));
+
+    const { post } = args;
+    
+        const getPostData = {
+            post
+        };
+
+    axios
+        .get("/posts", getPostData)
+        .then((res) => {
+            console.log(res.data);
+
+            const FBIdToken = `Bearer ${res.data.token}`;
+            // TODO: create a file in the user's machine to store the auth token
+            console.log(chalk.green.bold("you got the post successfully ✔️"));
+        })
+        .catch((error) => {
+            console.log(chalk.red.bold(`Could not obtain the post -> code: ${error.response.statusText}, message: ${JSON.stringify(error.response.data)}`));
+        });
+}
+//create post command definition
+const createPost = (args) => {
+
+    const { post, canLike, canUnlike, likes, tag } = args;
+
+    const postData = {
+        post,
+        canLike,
+        canUnlike,
+        likes,
+        tag
+    };
+
+    console.log(chalk.yellow.bold("posting..."));
+
+    axios
+        .post("/posts", postData)
+        .then((res) => {
+            console.log(res.data);
+
+            const FBIdToken = `Bearer ${res.data.token}`;
+            // TODO: create a file in the user's machine to store the auth token
+            console.log(chalk.green.bold("post created successfully ✔️"));
+        })
+        .catch((error) => {
+            console.log(chalk.red.bold(`Could not create the post -> code: ${error.response.statusText}, message: ${JSON.stringify(error.response.data)}`));
+        });
+}
+//update post command definition
+const updatePost = (args) => {
+
+    const { post, canLike, canUnlike, likes, tag } = args;
+
+    const putPostData = {
+        post,
+        canLike,
+        canUnlike,
+        likes,
+        tag
+    };
+
+    console.log(chalk.yellow.bold("editing the post..."));
+
+    axios
+        .put("/posts", putPostData)

```

Comment:
this is not really going to work. How can we update a particular post?

We need a parameter, the postId, so we know which post to modify exactly

Reply:
Done